ompi/mpi/fortran/mpif-h/cart_map_f.c

```diff
@@ -89,6 +90,10 @@ void ompi_cart_map_f(MPI_Fint *comm, MPI_Fint *ndims, MPI_Fint *dims,
     if (NULL != ierr) *ierr = OMPI_INT_2_FINT(c_ierr);
 
     OMPI_ARRAY_FINT_2_INT_CLEANUP(dims);
+#if OMPI_ALLOW_INPLACE_PATH_FOR_LOGICALS
```

Comment:
The related `C` binding is `MPI_Cart_create(..., const int periods[], ...)` so it looks like the real issue is we cannot convert `periods` back into Fortran, and the right fix here is to replace `OMPI_ARRAY_INT_2_LOGICAL(periods, size)` with `OMPI_ARRAY_LOGICAL_2_INT_CLEANUP(periods)`

The same comment applies to other parts of this PR.

make sense ?